R-package/src/xgboost_R.cc

```diff
@@ -413,9 +451,12 @@ XGB_DLL SEXP XGBoosterCreateInEmptyObj_R(SEXP dmats, SEXP R_handle) {
 
 XGB_DLL SEXP XGBoosterSetParam_R(SEXP handle, SEXP name, SEXP val) {
   R_API_BEGIN();
+  SEXP name_ = PROTECT(Rf_asChar(name));
+  SEXP val_ = PROTECT(Rf_asChar(val));
```

Comment:
Yes, that's also an option, but be aware that it doesn't play along with `rchk`: https://github.com/kalibera/rchk/issues/27

Since it seems `rchk` is not used in the CI jobs, maybe that's a good idea. I know DuckDB for example uses this type of pattern in their R interface.